src/controllers/auth.controller.js

```diff
@@ -0,0 +1,263 @@
+const bcrypt = require('bcrypt');
+const uuid = require('uuid');
+
+require('dotenv').config();
+
+const emailService = require('../services/email.service');
+const userService = require('../services/user.service');
+const jwtService = require('../services/jwt.service');
+const tokenService = require('../services/token.service');
+const { ApiError } = require('../utils/api.error');
+const { User } = require('../models/user.model');
+
+const register = async (req, res) => {
+  const { name, email, password } = req.body;
+  const activationToken = uuid.v4();
+
+  const { refreshToken } = req.cookies;
+  const userData = await jwtService.validateRefreshToken(refreshToken);
+
+  if (userData) {
+    throw ApiError.badRequest('Only for not authorized');
+  }
+
+  if (!name || !email || !password) {
+    throw ApiError.badRequest('Not all credentials are provided!');
+  }
+
+  const user = await userService.getByEmail(email);
+
+  if (user) {
+    throw ApiError.badRequest('The user with this email is already exists!');
```

Comment:
The error message 'The user with this email is already exists!' contains a grammatical error. It should be 'The user with this email already exists!'